src/site/content/en/blog/building-a-breadcrumbs-component/index.md

```diff
@@ -0,0 +1,498 @@
+  place-items: center;
+
+  & > * {
+    grid-area: stack;
+  }
+}
+```
+
+The `<select>` element is last in the DOM, so it's on top of the stack, and interactive. Add a style of `opacity: .01` so the element is still usable, and the result is a select box that fits perfectly to the shape of the icon. This is a nice way to customize the look of a select element while maintaining the built-in functionality.
+
+```css
+.disguised-select {
+  inline-size: 100%;
+  block-size: 100%;
+  opacity: .01;
+  font-size: 16px; /* fix iOS zoom */
+}
+```
+
+{% Aside %}
+Try using this icon component from mobile!
+{% endAside %}
+
+{% Video
+  src="video/vS06HQ1YTsbMKSFTIPl2iogUQP73/n3MsbELMwsvYn355hXdi.mp4",
+  class="w-screenshot",
+  autoplay="true",
+  loop="true",
+  muted="true"
+%}
+
+#### Overflow
+
+Breadcrumbs should be able to represent a very long trail. I'm a fan of allowing
+things to go offscreen horizontally, when appropriate, and I felt this
+breadcrumbs component qualified well. 
+
+```css
+.breadcrumbs {
+  overflow-x: auto;
+  overscroll-behavior-x: contain;
+  scroll-snap-type: x proximity;
+  scroll-padding-inline: calc(var(--nav-gap) / 2);
+
+  & > .crumb:last-of-type {
+    scroll-snap-align: end;
+  }
+
+  @supports (-webkit-hyphens:none) { & {
+    scroll-snap-type: none;
+  }}
+}
+```
+
+The overflow styles set up the following UX:
+
+1. Horizontal scroll with overscroll containment.
+1. Horizontal scroll padding.
+1. One snap point on the last crumb. This means that on page load the first
+   crumb loads snapped and in view.
+1. Removes the snap point from Safari, which struggles with the horizontal
+   scrolling and snap effect combinations.
+
+#### Media queries
+
+One subtle adjustment for smaller viewports is to hide the "Home" label, leaving
+just the icon: 
+
+```css
+@media (width <= 480px) { 
+  .breadcrumbs .home-label {
+    display: none;
+  }
+}
+```
+
+{% Img src="image/vS06HQ1YTsbMKSFTIPl2iogUQP73/T47A7AUqFPuneqNK3GWa.png",
+alt="side by side of the breadcrumbs with and without a home label, for
+comparison", width="800", height="84" %}
+
+## Accessibility
+
+### Motion
+
+There isn't a whole lot of motion in this component, but by wrapping the
+transition in a `prefers-reduced-motion` check, we can prevent unwanted motion.
+
+```css
+@media (prefers-reduced-motion: no-preference) { 
+  .crumbicon {
+    transition: box-shadow .2s ease;
+  }
+}
+```
+
+None of the other styles need to change, the hover and focus effects are great
+and meaningful without a `transition`, but if motion is OK, we'll add a subtle
+transition to the interaction.
+
+### Javascript
+
+Regardless of the type of router you use in your site or application, when a
+user interacts with the breadcrumbs, the URL needs to be updated, and the user
+shown the appropriate page. Two critical user experience measures are handled by
+Javascript here: select has changed and eager select change event firing
```

Comment:
```suggestion
JavaScript here: select has changed and eager select change event firing
```

Reply:
I suggest re-writing this bit, it wasn't clear to me what you're talking about until I read both the next paragraph and the code. Try explaining what issue you are trying to solve before naming the techniques. 

For example: "To provide good user experience, make sure no unexpected navigations happen when users are just browsing options...Select `changed` event should fire only when users select an option. Because some browsers fire this event while users are only navigating through options with a keyboard [do eager change event prevention]"

Reply:
re-written! 
thanks, with your help, i believe it's clearer 🙂